frontend/restInterfaces/SequenceRestInterface.ts

```diff
@@ -0,0 +1,129 @@
+import { BaseRestInterface } from './BaseRestInterface';
+import { ISlide } from '../../model/slide/ISlide';
+import { ISequence } from '../../model/sequence/ISequence';
+import { ISequenceWithSlides } from '../../model/sequence/ISequenceWithSlides';
+
+/**
+ * Implements communication with the Server concerning all Sequence-requests
+ */
+export class SequenceRestInterface extends BaseRestInterface {
+  /**
+   * Sends a request to backend to add a new Sequence
+   * @param title the title of the sequence the user is trying to add
+   * @returns code of the sequence
+   */
+  public static async addSequence(title: string): Promise<string> {
+    return await BaseRestInterface.put<string>('/sequence/', title);
+  }
+
+  /**
+   * Sends a request to backend to apply given changes to a sequence
+   * @param sequence the sequence object
+   * @returns confirmation
+   */
+  public static async updateSequence(
+    sequence: Partial<ISequenceWithSlides>
+  ): Promise<void> {
+    return await BaseRestInterface.patch('/sequence/', sequence);
+  }
+
+  /**
+   * Sends a request to backend to delete the given Sequence
+   * @param sequenceCode the code of the sequence
+   * @returns confirmation
+   */
+  public static async deleteSequence(sequenceCode: string): Promise<void> {
+    return await BaseRestInterface.delete('/sequence/', sequenceCode);
+  }
+
+  /**
+   * Sends a request to backend to get a Sequence by given Code
+   * @param sequenceCode the code of the sequence
+   * @returns requested sequence
+   */
+  public static async getSequence(
+    sequenceCode: string
+  ): Promise<ISequenceWithSlides> {
+    return BaseRestInterface.get<ISequenceWithSlides>(
+      `/sequence/${sequenceCode}/edit`
```

Comment:
I think we should leave this method uninplemented (throw an error that it is not implemented yet), since for it to properly work we would need the factories

Reply:
As far as I understood the axios generic methods (e.g. axios.get<T>) should do the job (i.e., converting the plain JSONs to types). Doesn't that work this way? Please refer to: https://upmostly.com/typescript/how-to-use-axios-in-your-typescript-apps for details.

Reply:
Partially yes, but the main issue here is that the things we need to cast are nested within ISequenceWithSlides (`slides.content`), and we need to distinguish which Content gets cast to which ContentType, since we only store `Content`, but need to cast them to `ImageContent`, `TextContent`,... based on their type attributes, which is done by the factories